File: backend/app/schemas/announcements.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
import datetime


class AnnouncementBase(BaseModel):
    text: str
    cta_text: Optional[str] = None
    cta_url: Optional[str] = None
    priority: Optional[str] = "normal"
    status: Optional[str] = "active"
    start_date: Optional[str] = None
    expiry_date: Optional[str] = None
    active: Optional[bool] = True


class AnnouncementCreate(AnnouncementBase):
    pass


class AnnouncementUpdate(BaseModel):
    text: Optional[str] = None
    cta_text: Optional[str] = None
    cta_url: Optional[str] = None
    priority: Optional[str] = None
    status: Optional[str] = None
    start_date: Optional[str] = None
    expiry_date: Optional[str] = None
    active: Optional[bool] = None


class AnnouncementResponse(AnnouncementBase):
    id: str
    created_at: Optional[datetime.datetime] = None
    updated_at: Optional[datetime.datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: src/api/announcements.ts
import { apiClient } from "./client"

export interface Announcement {
  id: string
  text: string
  ctaText?: string
  ctaUrl?: string
  cta_text?: string
  cta_url?: string
  priority: "high" | "normal" | "info"
  status: "active" | "draft" | "scheduled" | "expired"
  startDate?: string
  expiryDate?: string
  start_date?: string
  expiry_date?: string
  active: boolean
  createdAt?: string
  created_at?: string
  updatedAt?: string
}

export const announcementsApi = {
  getAll: () => apiClient<Announcement[]>("/api/announcements"),
  create: (data: Partial<Announcement>) =>
    apiClient<Announcement>("/api/announcements", {
      method: "POST",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  update: (id: string, data: Partial<Announcement>) =>
    apiClient<Announcement>(`/api/announcements/${id}`, {
      method: "PUT",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  delete: (id: string) =>
    apiClient<{ message: string; id: string }>(`/api/announcements/${id}`, {
      method: "DELETE",
      requireAuth: true,
    }),
}
